### ISSUE TITLE
Multiple default values do not work inside of blocks

### ISSUE THREAD

**Issue by alexander-schranz** [sulu/sulu#4987]
| Q | A
| --- | ---
| Sulu Version | 2.0.3
| Browser Version | -

# Actual Behavior

Currently the default value is only set on the last single_select.

<img width="518" alt="Bildschirmfoto 2020-01-17 um 15 00 50" src="https://user-images.githubusercontent.com/1698337/72617855-2c291e00-393a-11ea-8b8f-a15d0e61226f.png">

# Expected Behavior

Default value should be correct on both single_selects.

# Steps to Reproduce

<details>
<summary>Example Template</summary>

```xml
<?xml version="1.0" ?>
<template xmlns="http://schemas.sulu.io/template/template"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xmlns:xi="http://www.w3.org/2001/XInclude"
          xsi:schemaLocation="http://schemas.sulu.io/template/template http://schemas.sulu.io/template/template-1.0.xsd">

    <key>default</key>

    <view>pages/default</view>
    <controller>Sulu\Bundle\WebsiteBundle\Controller\DefaultController::indexAction</controller>
    <cacheLifetime>604800</cacheLifetime>

    <meta>
        <title lang="en">Default</title>
        <title lang="de">Standard</title>
    </meta>

    <properties>
        <property name="title" type="text_line" mandatory="true">
            <meta>
                <title lang="de">Titel</title>
                <title lang="en">Title</title>
            </meta>

            <params>
                <param name="headline" value="true"/>
            </params>

            <tag name="sulu.rlp.part"/>
        </property>

        <property name="url" type="resource_locator" mandatory="true">
            <meta>
                <title lang="de">Adresse</title>
                <title lang="en">Resourcelocator</title>
            </meta>

            <tag name="sulu.rlp"/>
        </property>

        <property name="description" type="text_editor">
            <meta>
                <title lang="en">Text</title>
                <title lang="de">Text</title>
            </meta>

            <tag name="sulu.search.field" role="description"/>
        </property>

        <block name="test" default-type="test">
            <types>
                <type name="test">
                    <properties>
                            <property name="position" type="single_select" colspan="6">
                            <meta>
                                <title lang="en">Position 1</title>
                                <title lang="nl">Positie 1</title>
                            </meta>
                            <params>
                                <param name="default_value" value="right"/>
                                <param name="values" type="collection">
                                    <param name="left">
                                        <meta>
                                            <title lang="en">Left</title>
                                            <title lang="nl">Links</title>
                                        </meta>
                                    </param>
                                    <param name="center">
                                        <meta>
                                            <title lang="en">Center</title>
                                            <title lang="nl">Midden</title>
                                        </meta>
                                    </param>
                                    <param name="right">
                                        <meta>
                                            <title lang="en">Right</title>
                                            <title lang="nl">Rechts</title>
                                        </meta>
                                    </param>
                                </param>
                            </params>
                        </property>

                        <property name="position2" type="single_select" colspan="6">
                            <meta>
                                <title lang="en">Position 2</title>
                                <title lang="nl">Positie 2</title>
                            </meta>
                            <params>
                                <param name="default_value" value="left"/>
                                <param name="values" type="collection">
                                    <param name="left">
                                        <meta>
                                            <title lang="en">Left</title>
                                            <title lang="nl">Links</title>
                                        </meta>
                                    </param>
                                    <param name="center">
                                        <meta>
                                            <title lang="en">Center</title>
                                            <title lang="nl">Midden</title>
                                        </meta>
                                    </param>
                                    <param name="right">
                                        <meta>
                                            <title lang="en">Right</title>
                                            <title lang="nl">Rechts</title>
                                        </meta>
                                    </param>
                                </param>
                            </params>
                        </property>
                    </properties>
                </type>
            </types>
        </block>

    </properties>
</template>
```

</details>

# Possible Solutions

Not sure

**Comment by luca-rath** [comment]
This problem occurs also with `select` properties.

**Comment by danrot** [comment]
I have investigated the problem, and realized that it is located in the [change handler of the blocks](https://github.com/sulu/sulu/blob/36278d0/src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FieldBlocks.js#L25). The `oldValues` variable in this code will still hold the initial value without any default after the first default value has been set. Then it will call the `onChange` callback the second time without the default value being set from the first property.

I currently see two solutions to this problem, whereby both of them are not completely satisfying.

1. We somehow merge the old values with the new ones instead of completely replacing the object. However, if we handle it like this I don't know how properties should be removed when we e.g. change to a different type.
2. I tried wrapping [the `onChange` call](https://github.com/sulu/sulu/blob/36278d0/src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FieldBlocks.js#L25) in a `setTimeout`. This will cause both of the `onChange` with the newer value, because execution is postponed. However, then we'll again have the problem that the [save button will be active immediately](https://github.com/sulu/sulu/issues/3798).

**Comment by alexandersch** [comment]
I am also experiencing this issue with checkboxes.

**Comment by Keimpy** [comment]
@danrot is there any update in this issue?

**Comment by danrot** [comment]
@Keimpy Sorry to say, but I didn't have any time to investigate that issue further :confused:

**Comment by danrot** [comment]
Fixed by #5576
